src/components/PostDetails.tsx

```diff
@@ -1,106 +1,99 @@
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
-
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
-            <div className="message-body" data-cy="CommentBody">
-              One more comment
-            </div>
-          </article>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+          {loading && <Loader />}
 
-            <div className="message-body" data-cy="CommentBody">
-              {'Multi\nline\ncomment'}
+          {error && (
+            <div className="notification is-danger" data-cy="CommentsError">
+              Something went wrong
             </div>
-          </article>
+          )}
+          {!error && !loading && (
+            <>
+              {comments.length > 0 && (
+                <>
+                  <p className="title is-4">Comments:</p>
+                  <CommentList
+                    comments={comments}
+                    onDelete={handleCommentDelete}
+                  />
+                </>
+              )}
 
-          <button
-            data-cy="WriteCommentButton"
-            type="button"
-            className="button is-link"
-          >
-            Write a comment
-          </button>
+              {!commenting && (
+                <button
+                  data-cy="WriteCommentButton"
+                  type="button"
+                  className="button is-link"
+                  onClick={() => setCommenting(true)}
+                >
+                  Write a comment
+                </button>
+              )}
+            </>
+          )}
         </div>
 
-        <NewCommentForm />
+        {!error && commenting && post && (
+          <NewCommentForm
+            post={post as Post}
```

Comment:
```suggestion
        {!error && commenting && post && (
          <NewCommentForm
            post={post}
```
assertions look redundant here, cause you render this component only when the post exists